src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java

```diff
@@ -396,26 +401,29 @@ public void dumpInventoryIntoWareHouse(@NotNull final InventoryCitizen inventory
      * @return the tile entity of the chest
      */
     @Nullable
-    private TileEntityChest searchRightChestForStack(@NotNull final ItemStack stack)
+    private IItemHandler searchRightItemHandler(@NotNull final ItemStack stack)
     {
-        if(InventoryUtils.findFirstSlotInInventoryWith(this, stack.getItem(), stack.getItemDamage()) != -1 && InventoryUtils.getOpenSlot(this) != -1)
+        if(InventoryUtils.findFirstSlotInInventoryWith(this.getItemHandler(), stack.getItem(), stack.getItemDamage()) != -1 && InventoryUtils.getOpenSlot(this.getItemHandler()) != -1)
         {
-            return this;
+            return this.getItemHandler();
         }
 
-        for(@NotNull final BlockPos pos : getBuilding().getAdditionalCountainers())
+        for(@NotNull final BlockPos pos : getBuilding().getAdditionalContainers())
         {
             final TileEntity entity = worldObj.getTileEntity(pos);
-            if(entity instanceof TileEntityChest
-                    && InventoryUtils.findFirstSlotInInventoryWith((TileEntityChest) entity, stack.getItem(), stack.getItemDamage()) != -1
-                    && InventoryUtils.getOpenSlot((TileEntityChest) entity) != -1)
+            if(entity instanceof TileEntityChest)
             {
-                return (TileEntityChest) entity;
+                IItemHandler handler = entity.getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, null);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'handler' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)